hummingbot/connector/exchange/latoken/latoken_exchange.py

```diff
@@ -0,0 +1,690 @@
+            return trade_update
+        exchange_order_id = trade["order"]
+        tracked_order = self._order_tracker.fetch_order(exchange_order_id=exchange_order_id)
+        if tracked_order is None:
+            return trade_update
+
+        trade_id = trade["id"]
+        fee = Decimal(trade["fee"])
+        price = Decimal(trade["price"])
+        quantity = Decimal(trade["quantity"])
+
+        spot_fee = TradeFeeBase.new_spot_fee(
+            fee_schema=self.trade_fee_schema(), trade_type=tracked_order.trade_type,
+            percent_token=tracked_order.quote_asset,
+            flat_fees=[TokenAmount(amount=fee, token=tracked_order.quote_asset)])
+        # This is a fill for a tracked order
+        trade_update = TradeUpdate(
+            trade_id=trade_id,
+            client_order_id=tracked_order.client_order_id,
+            exchange_order_id=exchange_order_id,
+            trading_pair=trading_pair,  # or tracked_order.trading_pair
+            fill_timestamp=int(timestamp),
+            fill_price=price,
+            fill_base_amount=quantity,
+            fill_quote_amount=Decimal(trade["cost"]),
+            fee=spot_fee,
+        )
+        return trade_update
+
+    def _db_tracked_order_candidate(self, trade, trading_pair: str):
+        # This is a fill of an order registered in the DB but not tracked any more
+        base_currency, quote_currency = trading_pair.split('-')
+        trade_type = TradeType.BUY if trade["makerBuyer"] else TradeType.SELL
+        order_type = OrderType.LIMIT
+        amount = Decimal(trade["quantity"])
+        price = Decimal(trade["price"])
+        trade_id = trade["id"]
+        exchange_order_id = trade["order"]
+        client_order_id = self._exchange_order_ids.get(exchange_order_id, None)
+        fee = Decimal(trade["fee"])
+        trade_fee = TradeFeeBase.new_spot_fee(
+            fee_schema=self.trade_fee_schema(), trade_type=trade_type,
+            percent_token=quote_currency,
+            flat_fees=[TokenAmount(amount=fee, token=quote_currency)])
+
+        # trade_fee = self.get_fee(base_currency, quote_currency, order_type, trade_type, amount, price, is_maker)
+        self._current_trade_fills.add(TradeFillOrderDetails(
+            market=self.display_name,
+            exchange_trade_id=trade_id,
+            symbol=trading_pair))
+
+        self.trigger_event(
+            MarketEvent.OrderFilled,
+            OrderFilledEvent(
+                timestamp=float(trade["timestamp"]) * 1e-3,
+                order_id=client_order_id,
+                trading_pair=trading_pair,
+                trade_type=trade_type,
+                order_type=order_type,
+                price=price,
+                amount=amount,
+                trade_fee=trade_fee,
+                exchange_trade_id=trade_id
+            ))
+        self.logger().info(f"Recreating missing trade in TradeFill: {trade}")
```

Comment:
Trade fills should be processed using the ClientOrderTracker, creating an instance of TradeUpdate, and sending it to the ClientOrderTracker.